query-service-impl/src/main/java/org/hypertrace/core/query/service/prometheus/QueryRequestToPromqlConverter.java

```diff
@@ -89,8 +107,10 @@ private QueryTimeRange getQueryTimeRange(ExecutionContext executionContext) {
                     && !QueryRequestUtil.isDateTimeFunction(expression))
         .map(
             functionExpression ->
-                mapToPromqlQuery(functionExpression, groupByList, filterList, duration))
-        .collect(Collectors.toUnmodifiableList());
+                ImmutablePair.of(
```

Comment:
nit: you're packing it into a pair and then immediately unpacking it.
1. You could use an `Map.Entry` to do the same thing, which is a more appropriate fit and doesn't introduce a third party
2. In this case, you don't need to have this map call at all, since the collector is already doing a key and value mapping You could just do
```java
.collect(Collectors.toMap(
  PrometheusUtils::getColumnNameForMetricFunction,
  function -> mapToPromqlQuery(functionExpression, groupByList, filterList, duration)
));
```

Reply:
yeah of course, not sure how this code got in